src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -385,10 +386,19 @@ def process_messages(
     strings = set()
     skipped_offsets = set()
     with metrics.timer("process_messages.parse_outer_message"):
-        parsed_payloads_by_offset = {
-            msg.offset: json.loads(msg.payload.value.decode("utf-8"), use_rapid_json=True)
-            for msg in outer_message.payload
-        }
+        parsed_payloads_by_offset: MutableMapping[int, json.JSONData] = {}
+        for msg in outer_message.payload:
+            try:
+                parsed_payload = json.loads(msg.payload.value.decode("utf-8"), use_rapid_json=True)
+                parsed_payloads_by_offset[msg.offset] = parsed_payload
+            except rapidjson.JSONDecodeError:
+                skipped_offsets.add(msg.offset)
```

Comment:
Could you explain what happens to the skipped offsets right now?

Reply:
@nikhars the `outer_message` that gets passed to this `process_message` function is made up of a bunch of messages (rn it's always 50). If we skip offsets, it means that when we are rebuilding the translated payload of messages it wouldn't have all the messages, the messages get dropped essentially. So `process_message` could return a batch of like 48 messages for example.